query.py

```diff
@@ -0,0 +1,394 @@
+import json
+import requests
+import networkx as nx
+
+
+class Query:
+    def __init__(
+        self,
+        service_base_url: str,
+        CapabilityStatement_path: str = None,
+    ) -> None:
+        self.base = service_base_url
+        self.possible_references = self._get_rev_include_possibilitie(
+            CapabilityStatement_path
+        )
+        """Instantiate the class and create the query object
+
+        Arguments:
+            service_base_url {str} -- the Service Base URL
+            CapabilityStatement_path {str} -- path to the json file that
+            contains a resource of type CapabilityStatement
+        """
+
+        # to represent the relationships (references) between resources
+        self.rscs_graph = nx.DiGraph()
+        self.rscs_type = dict()
+
+        self.main_internal_name = None
+        self.url_params = ""
+        self.url_rev_include = ""
+        self.count = None
+        self.api_url = None
+
+    def execute(
+        self,
+        select_dict: dict,
+        from_dict: dict,
+        join_dict: dict = None,
+        where_dict: dict = None,
+    ):
+        """Executes the query and returns (not defined yet?)
+
+        Arguments:
+            select_dict {dict} -- dictionary containing the attributes 
+            to be selected from the different resources
+            from_dict {dict} -- dictionary containing all requested 
+            resources
+
+        Keyword Arguments:
+            join_dict {dict} -- dictionary containing the inner join 
+            rules between resources (default: {None})
+            where_dict {dict} -- dictionary containing the (cumulative) 
+            conditions to be met by the resources (default: {None})
+        """
+        self._from(**from_dict)
+        if join_dict:
+            self._join(**join_dict)
+        if where_dict:
+            self._where(**where_dict)
+        self._select(**select_dict)
+
+        self.api_url = self._compute_url()
+
+        # to do :
+        # 1. retrieve the result of the request
+        # 2. Do all post-treatments
+
+    def from_config(self, config: dict):
+        """executes the query from a dictionary in the format of a 
+        configuration file
+
+        Arguments:
+            config {dict} -- dictionary in the format of a configuration
+            file
+        """
+        rename_keys = [
+            ("from", "from_dict"),
+            ("where", "where_dict"),
+            ("select", "select_dict"),
+            ("join", "join_dict"),
+        ]
+        for old_key, new_key in rename_keys:
+            try:
+                config[new_key] = config.pop(old_key)
+            except:
+                pass
+        self.execute(**config)
```

Comment:
Try to avoid this kind of manipulations when possible
Here, we could have
```
def from_config(self, config: dict):
    return self.execute(from_dict=config["from"], where_dict=config["where"], ...)
```

Reply:
Done :)